sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/filters/recurrence/RecurrenceEvaluatorTest.java

```diff
@@ -0,0 +1,534 @@
+    }
+
+    @Test
+    public void weeklyDayOfWeekIntervalFalse2() {
+        final ZonedDateTime now = ZonedDateTime.parse("2024-02-12T08:00:00+08:00");    // Monday in the 3rd week after the Start date
+        final TimeWindowFilterSettings settings = new TimeWindowFilterSettings();
+        final RecurrencePattern pattern = new RecurrencePattern();
+        final RecurrenceRange range = new RecurrenceRange();
+        final Recurrence recurrence = new Recurrence();
+        pattern.setType("Weekly");
+        pattern.setDaysOfWeek(List.of("Friday", "Monday"));
+        pattern.setFirstDayOfWeek("Sunday");
+        pattern.setInterval(2);
+        recurrence.setRange(range);
+        recurrence.setPattern(pattern);
+        settings.setStart("2024-02-02T12:00:00+08:00");    // Friday
+        settings.setEnd("2024-02-03T12:00:01+08:00");
+        settings.setRecurrence(recurrence);
+        consumeEvaluationTestData(settings, now, false);
+    }
+
+    @Test
+    public void weeklyDayOfWeekIntervalFalse3() {
+        final ZonedDateTime now = ZonedDateTime.parse("2023-09-04T00:00:00+08:00");    // Monday in the 2nd week after the Start date
+        final TimeWindowFilterSettings settings = new TimeWindowFilterSettings();
+        final RecurrencePattern pattern = new RecurrencePattern();
+        final RecurrenceRange range = new RecurrenceRange();
+        final Recurrence recurrence = new Recurrence();
+        pattern.setType("Weekly");
+        pattern.setDaysOfWeek(List.of("Monday", "Sunday"));
+        pattern.setFirstDayOfWeek("Monday");
+        pattern.setInterval(2);
+        recurrence.setRange(range);
+        recurrence.setPattern(pattern);
+        settings.setStart("2023-09-03T00:00:00+08:00");    // Sunday
+        settings.setEnd("2023-09-03T00:00:01+08:00");
+        settings.setRecurrence(recurrence);
+        consumeEvaluationTestData(settings, now, false);
+    }
+
+    @Test
+    public void weeklyDaysOfWeekNumberedRangeTrue() {
+        final ZonedDateTime now = ZonedDateTime.parse("2023-09-03T00:00:00+08:00");    // Third occurrence
+        final TimeWindowFilterSettings settings = new TimeWindowFilterSettings();
+        final RecurrencePattern pattern = new RecurrencePattern();
+        final RecurrenceRange range = new RecurrenceRange();
+        final Recurrence recurrence = new Recurrence();
+        pattern.setType("Weekly");
+        pattern.setDaysOfWeek(List.of("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"));
+        range.setType("Numbered");
+        range.setNumberOfRecurrences(3);
+        recurrence.setRange(range);
+        recurrence.setPattern(pattern);
+        settings.setStart("2023-09-01T00:00:00+08:00");    // Friday
+        settings.setEnd("2023-09-01T00:00:01+08:00");
+        settings.setRecurrence(recurrence);
+        consumeEvaluationTestData(settings, now, true);
+    }
+
+    @Test
+    public void weeklyDaysOfWeekNumberedRangeTrue2() {
```

Comment:
Could you add a opposite `weeklyDaysOfWeekNumberedRangeFalse` testcase for both existing `weeklyDaysOfWeekNumberedRangeTrue` testcases by substract one from the current `NumberOfOccurrences` and check whether it will return false with the same recurrence pattern.